packages/backend/src/apps/lettersg/helpers/process-missing-fields.ts

```diff
@@ -0,0 +1,34 @@
+import { IGlobalVariable } from '@plumber/types'
+
+export async function processMissingFields(
+  $: IGlobalVariable,
+): Promise<string[]> {
+  try {
+    const templateId = $.step.parameters.templateId as string
+    const { data } = await $.http.get('/v1/templates/:templateId', {
+      urlPathParams: {
+        templateId,
+      },
+    })
+
+    if (!data?.fields) {
+      return []
+    }
+    const allTemplateFields: string[] = data.fields
+    // follows subfield format for letterParams which is { field: string, value: string }
+    const inputTemplateFields = $.step.parameters.letterParams as Record<
+      string,
+      string
+    >[]
+
+    // perform filter for fields that don't exist
+    const inputFieldsSet: Set<string> = new Set()
+    for (const param of inputTemplateFields) {
+      inputFieldsSet.add(param['field'])
+    }
+    return allTemplateFields.filter((field) => !inputFieldsSet.has(field))
+  } catch (err) {
+    // not crucial so we don't alarm the user and return an empty array for the missing fields instead
+    return []
```

Comment:
can I check if this is desired behavior? should we not inform the user?

Reply:
yep i've discussed this with Wee Loong and we concluded to not show the missing fields if this API call fails unexpectedly because we won't know what error is thrown and this will be shown to the user which is going to confuse them further (since the user already knows its an issue with missing fields but just won't know which ones are missing in this context)